Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1473,27 +1507,29 @@ static void addUserInstalledFontURLs(NSString *path, Vector<URL>& fontURLs)
     if (!fontNames)
         return;
 
-    Vector<URL> fontURLs;
+    if (!m_userInstalledFontURLs) {
+        m_userInstalledFontURLs = { };
+        m_sandboxExtensionURLs = { };
+    }
 
-    for (NSString *fontName : fontNames) {
-        RetainPtr ctFont = adoptCF(CTFontCreateWithName(bridge_cast(fontName), 0.0, nullptr));
+    for (NSString *nsFontName : fontNames) {
+        RetainPtr ctFont = adoptCF(CTFontCreateWithName(bridge_cast(nsFontName), 0.0, nullptr));
         RetainPtr downloaded = adoptCF(static_cast<CFBooleanRef>(CTFontCopyAttribute(ctFont.get(), kCTFontDownloadedAttribute)));
         if (downloaded == kCFBooleanFalse)
             return;
         RetainPtr url = adoptCF(static_cast<CFURLRef>(CTFontCopyAttribute(ctFont.get(), kCTFontURLAttribute)));
-        fontURLs.append(URL(url.get()));
+        URL fontURL(url.get());
+        String fontName(nsFontName);
+        m_userInstalledFontURLs->set(fontName, fontURL);
```

Comment:
Does this really need to use `set()` instead of `add()`?

Reply:
I believe we can use `add` here.